File: medium/javascript/0151-reverse-words-in-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function (s) {
  const words = s.split(/\s+/);
  let result = "";

  for (let i = words.length - 1; i >= 0; i--) {
    result += words[i];
    result += " ";
  }

  return result.trim();
};

File: medium/go/0151-reverse-words-in-string.go
package main

import (
	"regexp"
	"strings"
)

func reverseWords(s string) string {
	reg := regexp.MustCompile("\\s+")
	words := reg.Split(s, -1)
	var result strings.Builder

	for i := len(words) - 1; i >=0; i-- {
		result.WriteString(words[i])
		result.WriteString(" ")
	}

	return strings.TrimSpace(result.String()) 
}